src/operation-node/set-operator-node.ts

```diff
@@ -0,0 +1,37 @@
+import { freeze } from '../util/object-utils.js'
+import { OperationNode } from './operation-node.js'
+import { RawNode } from './raw-node.js'
+import { SelectQueryNode } from './select-query-node.js'
+
+export type SetOperatorExpressionNode = SelectQueryNode | RawNode
+
+export type SetOperator = 'union' | 'interect' | 'except' | 'minus'
+
+export interface SetOperatorNode extends OperationNode {
```

Comment:
I'd rename this to `SetOperationNode` instead since it describes the whole operation and not just the operator. Otherwise all this looks great!

Reply:
Good point!

renamed.